### DISCUSSION TITLE
Handling geographical intersections

### DISCUSSION THREAD

**Discussion by vidhav** [directus/directus#19329] [Feature Requests]
### Summary

When looking at the source code for [_intersects](https://github.com/directus/directus/blob/cd2d601fbe3309ae54181c1a933d934d3d9afe3b/api/src/database/helpers/geometry/types.ts#L37) and [_intersects_bbox](https://github.com/directus/directus/blob/cd2d601fbe3309ae54181c1a933d934d3d9afe3b/api/src/database/helpers/geometry/types.ts#L50), it seems to me that they both do the same thing - send in any valid [GeoJSON geometry object](https://datatracker.ietf.org/doc/html/rfc7946#section-3.1) and if it intersects, you get a result.

This makes sense for the "intersects" filter, but a bbox implies that you have an actual [bounding box](https://wiki.openstreetmap.org/wiki/Bounding_Box) - in this case the coordinate of the lower left, and the upper right corners of the box: for Norway the bbox would be `[4.34028279580897, 57.7491183943321, 31.2191612226094, 71.2506114447295]`.

### Basic Example

```json
{
  "filter": {
    "geometry_field": {
      "_intersects_bbox": [4.34028279580897, 57.7491183943321, 31.2191612226094, 71.2506114447295]
    }
  }
}
```

### Motivation

- By treating `_intersects_bbox` as an actual bounding box, the meaning of the filter, and why it is different from `_intersects` becomes much clearer.
- This would also align Directus more closely with GIS standards.
- The database can query an actual bounding box instead of a square polygon.
  - Can improve performance.
  - Could be useful if Directus were to support other coordinate systems in the future (I've seen examples where the _fact_ that the earth is round, might produce some unwanted effects when using a polygon).

### Detailed Design

The bounding box consists of four numbers `xmin`, `ymin`, `xmax` and `ymax`. These should be four numbers in an array.

In PostGIS you can construct what's called an [envelope](https://postgis.net/docs/ST_MakeEnvelope.html) from these coordinates and then query the database:

```sql
SELECT * FROM "table" WHERE ST_Intersects("geometry_field", ST_MakeEnvelope(:xmin, :ymin, :xmax, :ymax, 4326));
SELECT * FROM "table" WHERE "geometry_field && ST_MakeEnvelope(:xmin, :ymin, :xmax, :ymax, 4326);
```

The first query is the usual intersection and second is something called a "bounding box overlaps" and is faster. Since we know we are dealing with bbox'es, PostGIS can at least speed things up by just comparing to the bbox of the stored geometries.

### Requirements List

None

### Drawbacks

- If someone is already querying `_intersects_bbox` with a GeoJSON geometry, this proposal will break.
- My GIS database skills are with PostGIS and there are probably other ways of doing this, in the other SQL dialects.

### Alternatives

I don't really see any alternatives when the docs say it's a bbox. There could be unwanted results when the query isn't doing what it should do in a GIS context.

### Adoption Strategy

As stated in the drawbacks, this will break for anyone who are querying the `_intersects_bbox` with a GeoJSON geometry. Information to the user is required.

- The `_intersects` still works the same way, so it can be used as a drop in replacement.
- Perhaps a guide in the docs could explain the difference between the two queries.

**Comment by michaeltoohig** [comment]
I concur, I expected this to use an actual bounding box.

For anyone wondering as I was, the current `_intersects` expects a GeoJSON Polygon.

```
const response = await directus.items('Land').readByQuery({
  limit: 100,
  fields: ['id', 'name', 'center', 'geometry'],
  filter: {
    center: {
      _intersects: {
        "type": "Feature",
        "geometry": {
          "coordinates": [
            [
              [
                168.2947501099543,
                -17.723682144590242
              ],
              [
                168.2947501099543,
                -17.727328428851507
              ],
              [
                168.29840874403044,
                -17.727328428851507
              ],
              [
                168.29840874403044,
                -17.723682144590242
              ],
              [
                168.2947501099543,
                -17.723682144590242
              ]
            ]
          ],
          "type": "Polygon"
        }
      }
    },
  },
});
```

**Comment by nickrum** [comment]
This is interesting because it seems like `_intersects_bbox` is indeed implemented using "boundling box overlaps" when using PostGIS over [here](https://github.com/directus/directus/blob/cd2d601fbe3309ae54181c1a933d934d3d9afe3b/api/src/database/helpers/geometry/dialects/postgres.ts#L17).
Also, as far as I understand, the idea of this operator is to be able to pass any geometry to it and the Database will calculate the bounding boxes and compare them to each other.
